tests/qa_automation/deploy_pods.pm

```diff
@@ -0,0 +1,69 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Installer for test scripts running inside the tested virtual machine
+# Maintainer: Martin Doucha <mdoucha@suse.cz>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Mojo::JSON;
+use Mojo::File 'path';
+
+sub run {
+    my ($self, $tinfo) = @_;
+    my $podlist = $tinfo->test;
+    my $path    = get_var('CASEDIR') . '/tests';
+
+    die "Pod list not provided" unless $podlist;
+
+    my $tardata = qx{tar -C "$path" -cjf - test_pods};
+    save_tmp_file('test_pods.tar.bz2', $tardata);
+    $tardata = undef;
+    $self->select_serial_terminal;
+    assert_script_run('pushd /tmp && rm -rf test_pods test_pods.tar.bz2');
+    assert_script_run('wget --quiet -O test_pods.tar.bz2 ' . autoinst_url . '/files/test_pods.tar.bz2');
+    assert_script_run('tar xjf test_pods.tar.bz2');
+    assert_script_run('rm test_pods.tar.bz2 && popd');
+
+    # Install pod dependencies
+    my %dep_hash = (perl => 1);
+
+    for my $podname (@$podlist) {
+        my $script = path("$path/test_pods/$podname")->slurp;
+        my @deplist;
+
+        if ($script =~ m/^#\s*pod_package_deps:\s*(.*)$/m) {
+            @deplist = split '\s*,\s*', $1;
+        }
+
+        for my $dep (@deplist) {
+            $dep_hash{$dep} = 1;
+        }
+    }
+
+    zypper_call 'in ' . join(' ', keys %dep_hash) if %dep_hash;
+    script_run("export _OPENQA_LOG_DIRECTORY='/tmp/openqa_logs'");
```

Comment:
I would simply use ˋassert_script_runˋ unless you check the exit code. Otherwise any failure can be near-silently skipped and cause weird results later one confusing test reviewers.

Reply:
Changed `script_run()` to `assert_script_run()` where it makes sense.